### PR TITLE
add BypassServiceClassCodeValidation

### PR DESCRIPTION
**Pull request by lukaszx0**
an extra validation option to skip SCC validation

### CODE REVIEW

**Review by prafmathur — APPROVED** [review]

File: batch.go [hunk 1]
```diff
@@ -1005,6 +1005,9 @@ func (batch *Batch) ValidTranCodeForServiceClassCode(entry *EntryDetail) error {
 
 	switch batch.Header.ServiceClassCode {
 	case AutomatedAccountingAdvices:
+		if batch.validateOpts.BypassServiceClassCodeValidation {
```
Comment (brian-column) [inline batch.go:1008]: if `batch.validateOpts != nil && batch.validateOpts.BypassServiceClassCodeValidation`

File: batchADV.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func (batch *BatchADV) Validate() error {
 	if batch.Header.StandardEntryClassCode != ADV {
 		return batch.Error("StandardEntryClassCode", ErrBatchSECType, ADV)
 	}
-	if batch.Header.ServiceClassCode != AutomatedAccountingAdvices {
+	if batch.Header.ServiceClassCode != AutomatedAccountingAdvices && !batch.validateOpts.BypassServiceClassCodeValidation {
```
Comment (brian-column) [inline batchADV.go:46]: ditto

File: batchARC.go [hunk 3]
```diff
@@ -60,8 +60,10 @@ func (batch *BatchARC) Validate() error {
 
 	// ARC detail entries can only be a debit, ServiceClassCode must allow debits
 	scc := batch.Header.ServiceClassCode
-	if scc == MixedDebitsAndCredits || (scc == CreditsOnly && !batch.IsReversal()) || (scc == DebitsOnly && batch.IsReversal()) {
-		return batch.Error("ServiceClassCode", ErrBatchServiceClassCode, scc)
+	if !batch.validateOpts.BypassServiceClassCodeValidation {
```
Comment (brian-column) [inline batchARC.go:63]: ditto

**Comment by brian-column** [comment]
by default, `batch.validateOpts = nil`

File: batch.go (hunk 1 above)
Comment (lukaszx0) [reply batch.go]: thx, added everywhere

**Review by brian-column — APPROVED** [review]

**Pull request merged**